### PR TITLE
use llama-2-7b by default

### PR DESCRIPTION
**Pull request by jeremymanning**
hey @HemuManju and @koerding,

I’ve been playing around with some options and prompts.  It’s a little slow on the free tier of colab (but still usable, I think!) and lets the user interact a little more fluidly with chatify.

there are three general options:
- explain what the given code does
- help debug the given code
- help fill in missing parts of the code
the responses are generated using a smaller version of the llama-2 model (llama-2-7b), which is free and fits in a free colab/kaggle instance.

to try it out, add and run the follow two cells to any tutorial notebook (including the comment in the second cell, and the two cells need to be separated, e.g., as opposed to combining them into a single cell):
```python
%pip install -q davos
import davos
davos.config.suppress_stdout = True
```
```python
smuggle chatify      # pip: git+https://github.com/jeremymanning/chatify.git
%load_ext chatify
```

then add the `%%explain` magic command to the top of any code cell to interact with the “tutor”.  you need to wait a bit after you click “submit request,” but it’s running!  i set it to stream tokens as they’re generated so that the user isn’t waiting with no feedback at all.

we could also replace this version with jordan’s server-based gpt-3.5 instance, or some other model running on a server (something we set up, hugging chat, chatgpt, etc.).  anything that can integrate with langchain will be pretty quick to set up and configure as the default.

what do you think of merging this change into the ContextLab (default) fork that students would see?  if we do that, we can stop worrying about the cache and both the comp neuro and DL courses would be shareable in their interactive forms.  i think the key questions are:
- are the long pauses where responses are generated so long that the entire thing isn’t usable?
- is the response quality sufficiently good that it’s useful?  from spot checking a few things the responses seem decent to me, but it’d be good to hear from others too

you may also want to check out the [prompts](https://github.com/jeremymanning/chatify/blob/main/chatify/prompts/tutor.yaml); to get them to generate nice outputs with llama-2 i needed to change them substantially.  i haven't done a ton of tweaking since the answers started to look reasonable, though.

### CODE REVIEW

**Comment by HemuManju** [comment]
@jeremymanning, awesome!! Thank you for this update 🥳!

File: chatify/main.py [hunk 1]
```diff
@@ -32,23 +34,37 @@ def __init__(self, shell=None, **kwargs):
...
                 open(pathlib.Path(str(dirname) + "/default_config.yaml")),
                 Loader=yaml.SafeLoader,
             )
-            print("Using default configuration!")
 
         self.prompts_config = self.cfg["prompts_config"]
         self.llm_chain = CreateLLMChain(self.cfg)
         self.tabs = None
 
+        # download local model if needed
```
Comment (HemuManju) [inline chatify/main.py:53]: @jeremymanning, I am not sure why we are deleting the LLM model? If we are deleting it which LLM model is used?

Comment (jeremymanning) [reply chatify/main.py:53]: it's just so the LLM instance isn't held in memory when the model is first loaded.  depending on which model is used, it could take up quite a large amount of RAM.  i included the "del llm" line so that the model is removed from memory (to free up RAM) until the %%explain magic command is used again.  but when it's called next, even though the model still has to take up lots of RAM (again, only for certain models), the model weights and other stuff that takes a long time to download will have already been made available from that first call.

in other words, the "point" of initializing the LLM when the chatify extension is first registered is to trigger the download of the model weights from the very start, instead of causing a long delay when the magic command is called for the first time.

Comment (jeremymanning) [reply chatify/main.py:53]: from playing around with it a bit, i felt like this tweak improved the user experience.  but let me know what you think!

Comment (HemuManju) [reply chatify/main.py:53]: Got it!

On another note, the [CreateLLMChain](https://github.com/ContextLab/chatify/blob/93b14c88785ffaf5c990d7e70486f770f7e80985/chatify/main.py#L49C1-L50C1) already performs an initialization of the LLM model. So we can skip the init part in the `main.py` function.

File: chatify/main.py [hunk 2]
```diff
@@ -32,23 +34,37 @@ def __init__(self, shell=None, **kwargs):
...
+        model_config = self.cfg["model_config"]
+        if model_config["model"] in ["huggingface_model", "llama_model"]:
+            print('Downloading model for local use; this may take a few minutes...')
+
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                if model_config["model"] == "huggingface_model":
+                    llm = HuggingFaceModel(model_config).init_model()
+                elif model_config["model"] == "llama_model":
+                    llm = LlamaModel(model_config).init_model()
```
Comment (jeremymanning) [inline chatify/main.py:63]: @HemuManju if I don't call the `.init_model()` functions, won't the "downloading model" part not run?

Comment (HemuManju) [reply chatify/main.py]: The `CreateLLMChain` class automatically calls it here: [models[model_].init_model()](https://github.com/ContextLab/chatify/blob/93b14c88785ffaf5c990d7e70486f770f7e80985/chatify/llm_models.py#L58)
